wallet/src/account/mod.rs

```diff
@@ -562,6 +562,124 @@ impl Account {
         .try_collect()
     }
 
+    pub fn sweep_addresses(
+        &mut self,
+        db_tx: &impl WalletStorageReadUnlocked,
+        destination: Destination,
+        request: SendRequest,
+        current_fee_rate: FeeRate,
+    ) -> WalletResult<SignedTransaction> {
+        let mut grouped_inputs = group_preselected_inputs(
+            &request,
+            current_fee_rate,
+            &self.chain_config,
+            self.account_info.best_block_height(),
+        )?;
+
+        let input_fees = grouped_inputs
+            .values()
+            .map(|(_, fee)| *fee)
+            .sum::<Option<Amount>>()
+            .ok_or(WalletError::OutputAmountOverflow)?;
+
+        let coin_input = grouped_inputs.remove(&Currency::Coin).ok_or(WalletError::NoUtxos)?;
+
+        let mut outputs = grouped_inputs
+            .into_iter()
+            .filter_map(|(currency, (amount, _))| {
+                let value = match currency {
+                    Currency::Coin => None?,
```

Comment:
Why not `return None`? This looks kinda weird... the try operator usually assumes a chance for success.